src/rgw/rgw_sal_motr.cc

```diff
@@ -1294,51 +1294,64 @@ MotrObject::~MotrObject() {
 //    return read_op.prepare(dpp);
 //  }
 
-int MotrObject::set_obj_attrs(const DoutPrefixProvider* dpp, RGWObjectCtx* rctx, Attrs* setattrs, Attrs* delattrs, optional_yield y, rgw_obj* target_obj)
+int MotrObject::fetch_obj_entry_and_key(const DoutPrefixProvider* dpp, rgw_bucket_dir_entry& ent, std::string& bname, std::string& key, rgw_obj* target_obj)
 {
-  // TODO : Set tags for multipart objects
-  if (this->category == RGWObjCategory::MultiMeta)
-    return 0;
+  ldpp_dout(dpp, 20) <<__func__<< ":" << "Entry" << dendl;
+
+  int rc = this->get_bucket_dir_ent(dpp, ent);
+  if (rc < 0) {
+      ldpp_dout(dpp, 0) <<__func__<< ":" << "ERROR: failed to get object entry. rc=" << rc << dendl;
+      return rc;
+    }
 
-  string bname, key;
   if (target_obj) {
     bname = get_bucket_name(target_obj->bucket.tenant, target_obj->bucket.name);
     key   = target_obj->key.to_str();
   } else {
     bname = get_bucket_name(this->get_bucket()->get_tenant(), this->get_bucket()->get_name());
     key   = this->get_key().to_str();
   }
-  ldpp_dout(dpp, 20) <<__func__<< ": " << bname << "/" << key << dendl;
+  ldpp_dout(dpp, 20) <<__func__<< ":" << bname << "/" << key << dendl;
 
-  // Get object's metadata (those stored in rgw_bucket_dir_entry).
-  bufferlist bl, update_bl;
-  string bucket_index_iname = "motr.rgw.bucket.index." + bname;
-  if (this->store->get_obj_meta_cache()->get(dpp, key, bl)) {
-    // Cache miss !
-    int rc = this->store->do_idx_op_by_name(bucket_index_iname, M0_IC_GET, key, bl);
-    if (rc < 0) {
-      ldpp_dout(dpp, 0) <<__func__<< ": Failed to get object's entry from bucket index. rc=" << rc << dendl;
+
+  // getting key for version cases
+  if(ent.key.instance == "null")
+  {
+    rc = this->fetch_null_obj_reference(dpp, key);
+    if (rc < 0)
       return rc;
-    }
+    ldpp_dout(dpp, 20) <<__func__<< ":" << "fetching null version key instance " << key << dendl;
+  } else {
+    key =  ent.key.name + "[" + ent.key.instance + "]";
+     ldpp_dout(dpp, 20) <<__func__<< ":" << "fetching null version key instance for latest object " << key << dendl;
   }
 
+   return 0;
+}
+
+int MotrObject::set_obj_attrs(const DoutPrefixProvider* dpp, RGWObjectCtx* rctx, Attrs* setattrs, Attrs* delattrs, optional_yield y, rgw_obj* target_obj)
+{
+  ldpp_dout(dpp, 20) <<__func__<< ":" <<"Entry" << dendl;
+
+  // TODO : Set tags for multipart objects
+  if (this->category == RGWObjCategory::MultiMeta)
+    return 0;
+
   rgw_bucket_dir_entry ent;
-  auto iter = bl.cbegin();
-  ent.decode(iter);
-  // Decoding current_attrs before MotrObject::Meta because 
-  // encode and decode always have to be sequential. 
-  rgw::sal::Attrs current_attrs;
```

Comment:
without this code, putObjectTagging is working still? this was added for same actually

Reply:
yes we made some changes yesterday. @selvakumaar5496 also agreed to same changes.

Reply:
done.

Reply:
alright